account_payment_sale/models/sale_order.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2014-2016 Akretion - Alexis de Lattre
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    payment_mode_id = fields.Many2one(
+        "account.payment.mode",
+        string="Payment Mode",
+        domain=[("payment_type", "=", "inbound")],
+    )
+
+    def _get_payment_mode_vals(self, vals):
+        if self.payment_mode_id:
+            vals["payment_mode_id"] = self.payment_mode_id.id
+            if self.payment_mode_id.bank_account_link == "fixed":
+                vals[
+                    "partner_bank_id"
+                ] = self.payment_mode_id.fixed_journal_id.bank_account_id.id
+        return vals
+
+    @api.onchange("partner_id")
+    def onchange_partner_id(self):
```

Comment:
Would be better to replace this onchange method with a computed method for field `payment_mode_id`.
So `payment_mode_id` should have: `computed="_compute_payment_mode_id"`, `readonly=False`, `store=True`.